ouroboros-network/src/Ouroboros/Network/Block.hs

```diff
@@ -108,27 +116,37 @@ castHash (BlockHash b) = BlockHash b
   Point on a chain
 -------------------------------------------------------------------------------}
 
--- | A point on the chain is identified by its 'Slot' and 'HeaderHash'.
---
--- The 'Slot' tells us where to look and the 'HeaderHash' either simply serves
--- as a check, or in some contexts it disambiguates blocks from different forks
--- that were in the same slot.
---
-data Point block = Point {
-       pointSlot :: SlotNo,
-       pointHash :: ChainHash block
-     }
-   deriving (Eq, Ord, Show)
+-- | A block is identified by a slot and a hash.
+data TBlockPoint slot hash = TBlockPoint
+  { pointSlot :: slot
+  , pointHash :: hash
+  }
+  deriving (Eq, Ord, Show)
+
+type BlockPoint block = TBlockPoint SlotNo (HeaderHash block)
```

Comment:
I'm a bit grumpy about having to have two different names here. I don't see why we need the T variant with its extra params. More names makes the API bigger and harder to remember and harder for new people to get into.

The `HeaderHash` is already a standalone type family, so surely it's already flexible enough. And it's not like there's many choices for the representation of `SlotNo`.